aten/src/ATen/native/quantized/cpu/qnnpack/src/q8dwconv/mp8x25-neon-per-channel.c

```diff
@@ -0,0 +1,939 @@
+        int32x4_t vaccX0_lo = vmull_s16(vget_low_s16(vxk0), vget_low_s16(vxi0));
+        int32x4_t vaccX0_hi =
+            vmull_s16(vget_high_s16(vxk0), vget_high_s16(vxi0));
+
+        const uint8x8_t vk1 = vld1_u8(w);
+        w += 8;
+        const uint8x8_t vi1 = vld1_u8(i1);
+        i1 += 8;
+        const int16x8_t vxk1 =
+            vreinterpretq_s16_u16(vsubl_u8(vk1, vkernel_zero_point));
+        const int16x8_t vxi1 =
+            vreinterpretq_s16_u16(vsubl_u8(vi1, vinput_zero_point));
+        int32x4_t vaccX1_lo = vmull_s16(vget_low_s16(vxk1), vget_low_s16(vxi1));
+        int32x4_t vaccX1_hi =
+            vmull_s16(vget_high_s16(vxk1), vget_high_s16(vxi1));
+
+        const uint8x8_t vk2 = vld1_u8(w);
+        w += 8;
+        const uint8x8_t vi2 = vld1_u8(i2);
+        i2 += 8;
+        const int16x8_t vxk2 =
+            vreinterpretq_s16_u16(vsubl_u8(vk2, vkernel_zero_point));
+        const int16x8_t vxi2 =
+            vreinterpretq_s16_u16(vsubl_u8(vi2, vinput_zero_point));
+        vaccX0_lo =
+            vmlal_s16(vaccX0_lo, vget_low_s16(vxk2), vget_low_s16(vxi2));
+        vaccX0_hi =
+            vmlal_s16(vaccX0_hi, vget_high_s16(vxk2), vget_high_s16(vxi2));
+
+        const uint8x8_t vk3 = vld1_u8(w);
+        w += 8;
+        const uint8x8_t vi3 = vld1_u8(i3);
+        i3 += 8;
+        const int16x8_t vxk3 =
+            vreinterpretq_s16_u16(vsubl_u8(vk3, vkernel_zero_point));
+        const int16x8_t vxi3 =
+            vreinterpretq_s16_u16(vsubl_u8(vi3, vinput_zero_point));
+        vaccX1_lo =
+            vmlal_s16(vaccX1_lo, vget_low_s16(vxk3), vget_low_s16(vxi3));
+        vaccX1_hi =
+            vmlal_s16(vaccX1_hi, vget_high_s16(vxk3), vget_high_s16(vxi3));
+
+        const uint8x8_t vk4 = vld1_u8(w);
+        w = (void*)((uintptr_t)w + sizeof(uint8x8_t));
+        const uint8x8_t vi4 = vld1_u8(i4);
+        i4 += 8;
+        const int16x8_t vxk4 =
+            vreinterpretq_s16_u16(vsubl_u8(vk4, vkernel_zero_point));
+        const int16x8_t vxi4 =
+            vreinterpretq_s16_u16(vsubl_u8(vi4, vinput_zero_point));
+        vaccX0_lo =
+            vmlal_s16(vaccX0_lo, vget_low_s16(vxk4), vget_low_s16(vxi4));
+        vaccX0_hi =
+            vmlal_s16(vaccX0_hi, vget_high_s16(vxk4), vget_high_s16(vxi4));
+
+        int32x4_t vacc_lo = vaddq_s32(vaccX0_lo, vaccX1_lo);
+        int32x4_t vacc_hi = vaddq_s32(vaccX0_hi, vaccX1_hi);
+
+        const int32x4_t vacc_lo_old = vld1q_s32(outacc);
+        outacc += 4;
+        const int32x4_t vacc_hi_old = vld1q_s32(outacc);
+        outacc += 4;
+        vacc_lo = vaddq_s32(vacc_lo, vacc_lo_old);
+        vacc_hi = vaddq_s32(vacc_hi, vacc_hi_old);
+
+        const float32x4_t requantization_scale_v_lo =
+            vld1q_f32(&quantization_params->neon.requantization_scales[channels - c]);
+        const float32x4_t requantization_scale_v_hi =
+            vld1q_f32(&quantization_params->neon.requantization_scales[channels - c + 4]);
```

Comment:
Likewise:

```
const float * const quantization_scales_channel = quantization_params->neon.requantization_scales + channels

for (...) {
  const float * const ptr = quantization_scales_channel - c;
  vld1q_f32(ptr)
  vld1q_f32(ptr + 4);
}
```

Only looking at the disassembly can prove whether this is a win.